### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Car0b1nius**
Adds type parameters and `@Override` annotations where appropriate, removes some unused imports, and fixes a few other things.

I considered combining `model.Node` and `robot.Node`, but I'd like to discuss that first.

### CODE REVIEW

File: ROVU/Core/src/model/Node.java [hunk 1]
```diff
@@ -73,7 +73,6 @@ public Node(boolean wall, boolean wifi, boolean eat, int id, Point p){
         this.wall = wall;
         this.wifi = wifi;
         this.eating = eat;
-        this.room = room;
```
Comment (elizaII) [inline ROVU/Core/src/model/Node.java]: Don't we need this in the future?

File: ROVU/Core/src/model/RewardCalculator.java [hunk 2]
```diff
@@ -1,13 +1,11 @@
 package model;
 
-import controller.DataHandler;
-import controller.RobotController;
```
Comment (elizaII) [inline ROVU/Core/src/model/RewardCalculator.java]: The `DataHandler` and the `RobotController` needs to stay there, the usage has been commented out in the meanwhile, but it's for future usage.

File: ROVU/Core/src/model/Node.java (hunk 1 above)
Comment (elizaII) [reply ROVU/Core/src/model/Node.java]: @nicket97 You defined `Node`, what do you think? Also, there's the problem of having `Node` defined in the package `Robot`, and the package `Model`. Are they two different types of nodes??

File: ROVU/Core/src/model/RewardCalculator.java (hunk 2 above)
Comment (Car0b1nius) [reply ROVU/Core/src/model/RewardCalculator.java]: Done, added a `@SuppressWarnings("unused")` instead.

File: ROVU/Core/src/model/Node.java (hunk 1 above)
Comment (Car0b1nius) [reply ROVU/Core/src/model/Node.java]: `Node.java` has been rewritten, and I've merged in those changes to this branch now.

**Pull request closed without merging**